### PR TITLE
Soll runtime test

### PR DESCRIPTION
**Pull request by eee4017**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -84,4 +84,21 @@ if(SOLL_INCLUDE_TESTS)
   include(Catch)
   add_subdirectory(unittests)
   add_subdirectory(test)
+  if(SOLL_RUNTIME_TESTS)
+      include(FetchContent)
```
Comment (hydai) [inline CMakeLists.txt:88]: The indent size is different from the original.

File: test/soll-runtime-test/evmc.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
+import ctypes
```
Comment (hydai) [inline test/soll-runtime-test/evmc.py:1]: We should use pylint or other python linter for formatting this file.

File: test/soll-runtime-test/runtest.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+SOLL_BASE="../.."
+
+cp ${SOLL_BASE}/build/utils/ssvm-evmc/test/soll-runtime-test/libsoll_runtime_test.so ${PWD}
+cp ${SOLL_BASE}/build/utils/ssvm-evmc/tools/ssvm-evmc/libssvm-evmc.so ${PWD}
+
+ln -sf ${SOLL_BASE}/test/libyul .
+ln -sf ${SOLL_BASE}/test/yul .
+
+python soll-runtime-test.py
```
Comment (hydai) [inline test/soll-runtime-test/runtest.sh:9]: Looks like you only copy and create new files without cleaning up after the `runtest.sh`?

File: test/soll-runtime-test/soll-runtime-test.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                print_green(f"--- storage check: PASS")
+            else:
+                print_red(f"--- storage check: FAIL")
+        elif num_testcase == 0:
+            num_run += 1
+            print(f"--- storage check: HAS NO TESTCASE") 
+        else:
+            print_red(f"{base}: FAIL")
+
+#print(num_test, num_run, num_has_testcase, num_pass)
```
Comment (hydai) [inline test/soll-runtime-test/soll-runtime-test.py:124]: Why do we have to keep this comment?

**Review by hydai — CHANGES_REQUESTED** [review]

File: test/soll-runtime-test/runtest.sh (hunk 3 above)
Comment (eee4017) [reply test/soll-runtime-test/runtest.sh:6]: the soft link file remains static and the shared library would be removed

File: test/soll-runtime-test/evmc.py (hunk 2 above)
Comment (eee4017) [reply test/soll-runtime-test/evmc.py:1]: I have applied https://github.com/hhatto/autopep8 on all python files

**Comment by hydai** [comment]
Please add a new job after `test soll`, to enable your runtime test:
https://github.com/second-state/SOLL/blob/master/.github/workflows/build.yml#L31

**Review by hydai — APPROVED** [review]

**Pull request merged**